### ISSUE TITLE
Internal TF error when changing a hyper-param

### ISSUE THREAD

**Issue by muh-bazm** [greatwhiz/tft_tf2#3]
Hi.
When I change the "total_time_steps" parameter to any number smaller than the default value, I get the following error:

----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
tensorflow.python.framework.errors_impl.InternalError: 2 root error(s) found.
  (0) INTERNAL: No algorithm worked!
	 [[{{node TemporalFusionTransformer/cu_dnnlstm_1/CudnnRNNV2}}]]
	 [[Func/training/Adam/gradients/gradients/TemporalFusionTransformer/dropout_16/cond_grad/StatelessIf/then/_550/input/_1652/_1919]]
  (1) INTERNAL: No algorithm worked!
	 [[{{node TemporalFusionTransformer/cu_dnnlstm_1/CudnnRNNV2}}]]
0 successful operations.
0 derived errors ignored.
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------

my machine:
device: Lenovo ideapad 320
OS: Ubuntu 20.04
CPU: corei7 7th gen
Memory: 20 GB
GPU: NVIDIA GeForce 920MX

**Comment by greatwhiz** [comment]
It seems like a common memory problem.

Please try to add this code at the begining:
`tf.config.experimental.set_memory_growth(physical_devices[0], True)`

Reference:
https://stackoverflow.com/questions/59340465/how-to-solve-no-algorithm-worked-keras-error
https://github.com/tensorflow/tensorflow/issues/43174

**Comment by muh-bazm** [comment]
Thank you for your advice. I added this line of code, but it did not change the outcome. Finally, I could solve the problem by changing another hyper-parameter. First, I was interested in changing "total_time_steps" hyper-parameter. However, later I notice the value for "total_time_steps" cannot be smaller than "num_encoder_steps." So I assigned a value to the latter less than the former hyper-parameter. Problem solved. 

Anyway, thank you very much for you help.
